axelor-base/src/main/resources/views/Partner.xml

```diff
@@ -727,7 +727,9 @@
 	</action-attrs>
 
 	<action-record name="action-partner-record-set-balance" model="com.axelor.apps.base.db.Partner">
-		<field name="$balance" expr="eval: __repo__(AccountingSituation).all().filter('self.partner.id = ?', id).fetch().sum{ it.balanceDueCustAccount }"/>
+        <field name="$balance" expr="eval: __config__.app.getApp('account')?.partnerBalanceViewerConfiguration == 1 ? __repo__(AccountingSituation).all().filter('self.partner.id = ?', id).fetch().sum{ it.balanceCustAccount } : __config__.app.getApp('account')?.partnerBalanceViewerConfiguration == 2 ?
+        __repo__(AccountingSituation).all().filter('self.partner.id = ?', id).fetch().sum{ it.balanceDueCustAccount } :
+        __repo__(AccountingSituation).all().filter('self.partner.id = ?', id).fetch().sum{ it.balanceDueDebtRecoveryCustAccount }"/>
```

Comment:
On Partner, should be done in JEL not groovy to be consistent.

Reply:
There is no sum function in JEL. Should we create a service to be called to do the sum ?